Tavisca.Bootcamp.LanguageBasics.Exercise2/Program.cs

```diff
@@ -2,29 +2,95 @@
             Test(new[] { "11:59:13", "11:13:23", "12:25:15" }, new[] { "few seconds ago", "46 minutes ago", "23 hours ago" }, "11:59:23");
             Console.ReadKey(true);
         }
 
-        private static void Test(string[] postTimes, string[] showTimes, string expected)
-        {
+        private static void Test(string[] postTimes, string[] showTimes, string expected)
+        {
             var result = GetCurrentTime(postTimes, showTimes).Equals(expected) ? "PASS" : "FAIL";
             var postTimesCsv = string.Join(", ", postTimes);
             var showTimesCsv = string.Join(", ", showTimes);
-            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
+            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
         }
 
-        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
+        {
+            var currentTimeLowerBounds = new DateTime[exactPostTime.Length];
+            var currentTimeUpperBounds = new DateTime[exactPostTime.Length];
+
+            for (int i = 0; i < exactPostTime.Length; i++)
+            {
+                var exactTime = DateTime.Parse(exactPostTime[i], System.Globalization.CultureInfo.InvariantCulture);
+                var currentTimeLowerBound = exactTime;
+                var currentTimeUpperBound = exactTime;
+                var displayedPostTimeComponents = showPostTime[i].Split(' ');
+
+                //Determining possible corresponding upper bound and lower vound values of current time.
+                switch (displayedPostTimeComponents[1])
+                {
+                    case "seconds":
+                        currentTimeUpperBound = exactTime.AddSeconds(59);
+                        break;
+
+                    case "minutes":
+                        currentTimeLowerBound = exactTime.AddMinutes(double.Parse(displayedPostTimeComponents[0]));
+                        currentTimeUpperBound = currentTimeLowerBound.AddSeconds(59);
+                        break;
+
+                    case "hours":
+                        currentTimeLowerBound = exactTime.AddHours(double.Parse(displayedPostTimeComponents[0]));
+                        currentTimeUpperBound = currentTimeLowerBound.AddMinutes(59);
+                        currentTimeUpperBound = currentTimeLowerBound.AddSeconds(59);
+                        break;
+
+                    default:
+                        break;
+                }
+
+                currentTimeLowerBounds[i] = currentTimeLowerBound;
+                currentTimeUpperBounds[i] = currentTimeUpperBound;
+            }
+
+            for (int i = 0; i < exactPostTime.Length - 1; i++)
+            {

```

Comment:
Instead of 2 loops, you can do it with a single loop, given that the length of both the args are same. 
Also, such terminal checks should be performed before starting the actual logic of the operation, i.e. at the start of the method.

Reply:
Changed the loop structure.
Shifted terminal logic to the top.